core/trino-spi/src/main/java/io/trino/spi/statistics/DoubleRange.java

```diff
@@ -42,7 +42,7 @@ public static Optional<DoubleRange> from(Type type, Object minTrinoNativeValue,
         OptionalDouble min = toStatsRepresentation(type, minTrinoNativeValue);
         OptionalDouble max = toStatsRepresentation(type, maxTrinoNativeValue);
 
-        if (min.isEmpty() && max.isEmpty()) {
+        if (min.isEmpty() && max.isEmpty() || min.isPresent() && Double.isNaN(min.getAsDouble()) || max.isPresent() && Double.isNaN(max.getAsDouble())) {
```

Comment:
Move the new condition to after line 54, then the isPresent checks are not needed

Reply:
Fixed, thanks for the review